### PR TITLE
Update sparse semi-structured linear operator

### PR DESCRIPTION
**Pull request by alexsamardzic**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #104608
* #103830

cc @nikitaved @pearu @cpuhrsch @amjames @bhosmer

### CODE REVIEW

**Comment by alexsamardzic** [comment]
Just removed the code doing dense->(sparse,meta) conversion, thus sparse semi-structured linear operator simplified (renamed, for consistency, too).

File: docs/source/sparse.rst [hunk 1]
```diff
@@ -300,7 +300,7 @@ To use these ops, simply pass the output of ``to_sparse_semi_structured(tensor)`
     >>> torch.allclose(c, torch.mm(a_sparse, b))
     True
 
-Under the hood, SparseSemiStructuredTensor will call ``torch._structured_sparse_linear`` for accelerated inference using CUTLASS sparse kernels.
```
Comment (cpuhrsch) [inline docs/source/sparse.rst:303]: We can remove this entirely. We don't document private functions.

**Review by cpuhrsch — APPROVED** [review]
See small comments, otherwise this is fine.

File: docs/source/sparse.rst (hunk 1 above)
Comment (alexsamardzic) [reply docs/source/sparse.rst:303]: Fixed.

**Pull request closed without merging**